### ISSUE TITLE
BUG: Line comments comments in indented if-block are unindented

### ISSUE THREAD

**Issue by corneliusroemer** [snakemake/snakefmt#161]
In v0.7.0 (and also 0.6.1) line comments starting with `#` in an indented if-block are unindented by Snakemake.

While this does not change the execution of the code, it is not how one would want comments to be formatted in general.

```py
if config.get("s3_dst"):

    rule one:
        output:
            touch("data/one.txt"),

    # This gets falsely unindented
    rule two:
        output:
            touch("data/two.txt"),
```

becomes

```py
if config.get("s3_dst"):

    rule one:
        output:
            touch("data/one.txt"),

# This gets falsely unindented
    rule two:
        output:
            touch("data/two.txt"),
```

**Comment by corneliusroemer** [comment]
This bug also happens with comments inside an indented rule: the comment gets totally unindented and then snakefmt thinks the rule has nothing associated with it - which is obviously not true since the code executes just fine:

From this:
```py
if config.get("s3_dst"):

    rule one:
        #Test
        output:
            touch("s3_dst")
```

Snakefmt makes this:
```py
if True:

    rule one:
# Test
        output:
            touch("s3_dst"),
```

Which it then red-squiggles:
<img width="1098" alt="image" src="https://user-images.githubusercontent.com/25161793/200713859-2cee447d-172d-4e44-a7a0-c86b6ff7a69b.png">

**Comment by mbhall88** [comment]
Thank you for all of the issue reports @corneliusroemer.

Would you be able to do me a favour and see if these bugs were introduced in v0.7.0? i.e. could you test you snakefile with v0.61?

**Comment by mbhall88** [comment]
Okay, thanks for that!

We'll try and get this fixed as quick as we can. However, we may be a little slow as one maintainer has his PhD viva this week and I am going on holidays for a week in a few days time.

**Comment by mbhall88** [comment]
This bug still seems to exist in v0.8.0 sorry. We'll get to it soon

**Comment by mbhall88** [comment]
This is resolved in v0.8.0 from my testing
